### PR TITLE
[zh-cn]: update types_of_attacks

### PR DESCRIPTION
**Pull request by Ende93**
### Description

[zh-cn]: update types_of_attacks

### CODE REVIEW

File: files/zh-cn/web/security/types_of_attacks/index.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
+---
+title: 攻击类型
+slug: Web/Security/Types_of_attacks
```
Comment (jasonren0403) [inline files/zh-cn/web/security/types_of_attacks/index.md:3]: Please add [l10n.sourceCommit](https://github.com/mdn/translated-content/blob/main/docs/zh-cn/translation-guide.md#元数据) key

File: files/zh-cn/web/security/types_of_attacks/index.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+---
+
+{{QuickLinksWithSubpages("/en-US/docs/Web/Security")}}
```
Comment (jasonren0403) [inline files/zh-cn/web/security/types_of_attacks/index.md:6]: To display sidebar properly

```suggestion
{{QuickLinksWithSubpages("/zh-CN/docs/Web/Security")}}
```

File: files/zh-cn/web/security/types_of_attacks/index.md [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+CSRF（有时也称为 XSRF）是一类相关的攻击。攻击者使用户的浏览器在用户不知情的情况下向网站的后端发送请求。攻击者可以使用 XSS 载荷发起 CSRF 攻击。
+
+维基百科提到了一个很好的 CSRF 示例。在这种情况下，某人访问了一个实际上并不是图像的图像（例如在未经过滤的聊天或论坛中），而是向银行服务器发出取款请求：
+
+```html
+<img
+  src="https://bank.example.com/withdraw?account=bob&amount=1000000&for=mallory" />
+```
+
+现在如果你登录银行帐户并且你的 cookie 仍然有效（并且没有其他验证），那么在加载包含此图像的 HTML 后你将立即转账。对于需要 POST 请求的端点，可以在加载页面时以编程方式触发 \<form> 提交（可能在不可见的 \<iframe> 中）：
```
Comment (jasonren0403) [inline files/zh-cn/web/security/types_of_attacks/index.md:44]:
```suggestion
现在如果你登录银行账户并且你的 cookie 仍然有效（并且没有其他验证），那么在加载包含此图像的 HTML 后你将立即转账。对于需要 POST 请求的端点，可以在加载页面时以编程方式触发 \<form> 提交（可能在不可见的 \<iframe> 中）：
```

File: files/zh-cn/web/security/types_of_attacks/index.md [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+  });
+</script>
+```
+
+下面有一些技术可以用来防止这种情况发生：
+
+- GET 端点应该是幂等的——执行更改但不检索数据的操作应该要发送 POST（或其他 HTTP 方法）请求。POST 端点不应接受带有查询字符串参数的 GET 请求。
+- 服务器应向浏览器提供会话唯一的 CSRF 令牌。然后，只要浏览器提交表单（在 `<form>` 元素的隐藏输入字段中），就可以包含此令牌。对于所有可能执行操作的非 GET 请求，服务器会将接收到的令牌与其存储的会话值进行比较。如果令牌不匹配，请求将被中止。
+- 这种保护方法依赖于攻击者无法预测用户分配的 CSRF 令牌。应在登录时重新生成令牌。
+- 用于敏感操作的 Cookie（例如会话 Cookie）应具有较短的生命周期，并且 SameSite 属性设置为 Strict 或 Lax。 （参见 [SameSite](/zh-CN/docs/Web/HTTP/Cookies#samesite_attribute) cookie 属性）。在支持的浏览器中，这将确保会话 cookie 不会随跨站请求一起发送，因此对应用程序服务器来说这个请求是未经验证的。
```
Comment (jasonren0403) [inline files/zh-cn/web/security/types_of_attacks/index.md:64]: redundant space

```suggestion
- 用于敏感操作的 Cookie（例如会话 Cookie）应具有较短的生命周期，并且 SameSite 属性设置为 Strict 或 Lax。（参见 [SameSite](/zh-CN/docs/Web/HTTP/Cookies#samesite_attribute) cookie 属性）。在支持的浏览器中，这将确保会话 cookie 不会随跨站请求一起发送，因此对应用程序服务器来说这个请求是未经验证的。
```

File: files/zh-cn/web/security/types_of_attacks/index.md [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+
+会话劫持是指获取并滥用用户的已验证会话。这可能是通过窃取现有会话的 cookie，或者通过欺骗用户（或其浏览器）设置可预测的会话 ID 的 cookie 来实现的。
+
+通过部署严格的内容安全策略（Content-Security-Policy），可以限制数据泄露途径。
+
+### 会话固定
+
+第三方能够确定用户的会话标识符（例如，通过读取或设置它），因此可以作为用户与服务器进行交互。窃取 cookie 是实现这一目的的一种方法。
+
+请注意，子域名（例如 application.example.com）可以通过设置 Domain 属性来设置一个 cookie，以便在发送请求到 example.com 或其他子域名时将其发送出去：
```
Comment (jasonren0403) [inline files/zh-cn/web/security/types_of_attacks/index.md:83]:
```suggestion
请注意，子域名（例如 application.example.com）可以通过设置 `Domain` 属性来设置一个 cookie，以便在发送请求到 example.com 或其他子域名时将其发送出去：
```

File: files/zh-cn/web/security/types_of_attacks/index.md [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+
+```http
+Set-Cookie: CSRF=e8b667; Secure; Domain=example.com
+```
+
+如果子域上存在易受攻击的应用程序，那么此机制就会被会话固定攻击滥用。当用户访问父域（或另一个子域）上的页面时，应用程序可能会信任用户 cookie 中发送的现有值。这可能允许攻击者绕过 CSRF 保护或在用户登录后劫持会话。或者，如果父域不使用带有 `includeSubdomains` 设置的 [HTTP Strict-Transport-Security](/zh-CN/docs/Glossary/HSTS)，则用户将受到活动 MitM 的约束（可能连接到公共 Wi-Fi 网络）可能会收到来自不存在的子域的带有 Set-Cookie 标头的响应。最终结果大致相同，浏览器存储了非法 cookie 并将其发送到 example.com 下的所有其他页面。
```
Comment (jasonren0403) [inline files/zh-cn/web/security/types_of_attacks/index.md:89]:
```suggestion
如果子域上存在易受攻击的应用程序，那么此机制就会被会话固定攻击滥用。当用户访问父域（或另一个子域）上的页面时，应用程序可能会信任用户 cookie 中发送的现有值。这可能允许攻击者绕过 CSRF 保护或在用户登录后劫持会话。或者，如果父域不使用带有 `includeSubdomains` 设置的 [HTTP Strict-Transport-Security](/zh-CN/docs/Glossary/HSTS)，则受到主动 MitM 攻击的用户（可能连接到公共 Wi-Fi 网络）可能会收到来自不存在的子域的带有 Set-Cookie 标头的响应。最终结果大致相同，浏览器存储了非法 cookie 并将其发送到 example.com 下的所有其他页面。
```

**Review by jasonren0403 — CHANGES_REQUESTED** [review]
Here are some proposed changes

**Review by Ende93 — COMMENTED** [review]
resolved

**Review by jasonren0403 — APPROVED** [review]
LGTM, Thank you @Ende93 👍

**Pull request merged**